filtering/cel.go

```diff
@@ -34,20 +36,86 @@ type CELFilter struct {
 	valueWhenNoop bool
 }
 
+type blocknumBasedCELFilter map[uint64]*CELFilter
+
+func (bbcf blocknumBasedCELFilter) String() (out string) {
+	if len(bbcf) == 1 {
+		for _, v := range bbcf {
+			return v.code
+		}
+	}
+	var arr []uint64
+	for k := range bbcf {
+		arr = append(arr, k)
+	}
+	sort.Slice(arr, func(i int, j int) bool { return arr[i] < arr[j] })
+	for _, k := range arr {
+		out += fmt.Sprintf("#%d;%s", k, bbcf[k].code)
+	}
+	return
+}
+
+func (bbcf blocknumBasedCELFilter) choose(blknum uint64) *CELFilter {
+	var highestMatchingKey uint64
+	for k := range bbcf {
+		if blknum >= k && k > highestMatchingKey {
+			highestMatchingKey = k
+		}
+	}
+	if v, ok := bbcf[highestMatchingKey]; ok {
+		return v
+	}
```

Comment:
Also, what happens if there is no filter define for the block range? For example someone trying to do `#10; aaa;;; #20; bbb`, this will return a `nil` filter.

It would be preferable to return the "noop" CEL filter for the given element.

Reply:
I cannot return it directly because I am iterating on a map, the keys are unordered and I have to finish iterating on all values.
ok for the "noop", I'll find how to do it

Reply:
It's not directly, it's at the end of the loop. It's just to save the extra lookup at the end.

Reply:
noop filters are either `true` or `false` depending on the direction, no?

weren't we doing `#10@` ? and not `#10;` ?

Reply:
fixed in last commit, and we were doing #10; https://github.com/dfuse-io/dfuse-eosio/issues/129#issuecomment-681996973